types/lodash/index.d.ts

```diff
@@ -12059,7 +12063,22 @@ declare namespace _ {
          * _.isArrayLike(_.noop);
          * // => false
          */
-        isArrayLike<T>(value?: any): value is T[];
+        isArrayLike<T>(value: T & string & number): boolean; // should only match if T = any
```

Comment:
Once we get [sutraction types](https://github.com/Microsoft/TypeScript/issues/4183) this should become a lot easier.